src/chunkserver/chunkserver-common/cmr_disk.cc

```diff
@@ -173,17 +180,37 @@ void CmrDisk::open(IChunk *chunk) {
 }
 
 int CmrDisk::unlinkChunk(IChunk *chunk) {
-	int result = 0;
+	// Get absolute paths for meta and data files
+	const std::filesystem::path metaFile = chunk->metaFilename();
+	const std::filesystem::path dataFile = chunk->dataFilename();
+
+	// Use the metaPath() and dataPath() to get the disk paths
+	const std::string metaDiskPath = metaPath();
+	const std::string dataDiskPath = dataPath();
+
+	// Ensure we found a valid disk path
+	if (metaDiskPath.empty() || dataDiskPath.empty()) {
+		safs_pretty_errlog(LOG_ERR, "Error finding disk path for chunk: %s",
+		                   chunk->metaFilename().c_str());
+		return SAUNAFS_ERROR_ENOENT;
+	}
+
+	// Create a deletion timestamp
+	const std::string deletionTime = ChunkTrashManager::getDeletionTimeString();
 
-	if (::unlink(chunk->metaFilename().c_str()) != 0) {
-		result = -1;
+	// Move meta file to trash
+	int result = ChunkTrashManager::moveToTrash(metaFile, metaDiskPath, deletionTime);
+	if (result != SAUNAFS_STATUS_OK) {
+		return result;
 	}
 
-	if (::unlink(chunk->dataFilename().c_str()) != 0) {
-		result = -1;
+	// Move data file to trash
+	result = ChunkTrashManager::moveToTrash(dataFile, dataDiskPath, deletionTime);
+	if (result != SAUNAFS_STATUS_OK) {
+		return result;
 	}
 
-	return result;
+	return SAUNAFS_STATUS_OK;
```

Comment:
Consider adding a config option regarding the new behavior and old behavior, with the old behaviour being default. This is quite a change in behavior and can cause confusion and unexpected results ("why is chunkserver not deleting chunks?") for existing installations.

Reply:
Under normal  circumstances, I don't believe the older deletion operation is inherently unsafe. The two main causes of accidental deletions have been 1) Upgrades to newer versions and 2) Migrations from LizardFS, both of which are unusual (not a day-to-day activity).

The other thing is that in general, new features should be disabled by default, not only because they haven't been field-tested but also because they might interfere with existing functionality, in this case the deletion of chunks. When bumping major versions we can revisit whether something should be enabled by default.

However, the point of it being safer in general is persuasive enough that I'd go forgo the above and have it enabled by default, but only when the feature is on par with the old behavior (it actually eventually deletes the chunks). The main idea being we should not break existing setups that rely on the chunks actually being deleted. 

Once the feature is fully-complete, we could come back and decide to make it on by default. Until then, we need to inform users/packagers that this is an option, that we encourage usage of it (but they need to be aware that it's not deleting chunks automatically, they have to do it manually) and let them decide whether to use it or not.